src/main/java/htsjdk/samtools/cram/build/CRAMReferenceRegion.java

```diff
@@ -113,9 +113,14 @@ public void fetchReferenceBases(final int referenceIndex) {
 
         // Re-resolve the reference bases if we don't have a current region or if the region we have
         // doesn't span the *entire* contig requested.
+        final SAMSequenceRecord newSequenceRecord = sequenceDictionary.getSequence(referenceIndex);
+        if (newSequenceRecord == null) {
+            throw new IllegalArgumentException(
+                    String.format("Requested reference sequence index %d not found", referenceIndex));
+        }
```

Comment:
Can you call the existing `getSAMSequenceRecord()` method here? It does exactly the same thing.

Reply:
Oh yeah, done.